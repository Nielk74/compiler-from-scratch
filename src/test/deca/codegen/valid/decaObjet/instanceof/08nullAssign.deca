// Description:
//    assignation du résultat d'un instanceof avec null à une variable boolean
//
// Resultats:
//    null is instance of A
//
// Historique:
//    cree le 17/06/2022

class A {}

{
    boolean b = null instanceof A;
    if (b) {
        println("null is instance of A");
    } else {
        println("null is not instance of A");
    }
}