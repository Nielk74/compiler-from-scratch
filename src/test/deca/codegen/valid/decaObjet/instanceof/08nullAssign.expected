null is instance of A
